pathway-analysis/kegg_ortholog_mapping.Rmd

```diff
@@ -0,0 +1,252 @@
+mouse_ortholog_df <- entrez_in_pathways_df %>% 
+  dplyr::inner_join(mouse, by = "human_entrez")
+```
+
+### 1:many mapping
+
+For 1:many mappings, we'll pick the one with the most support.
+Here, we'll only consider the _number_ of sources without any regard for _what_
+those resources are.
+Essentially, we weigh all sources equally though they likely have more or less
+permissive criteria.
+
+```{r}
+# add a column that counts resources
+mouse_ortholog_df <- mouse_ortholog_df %>%
+  dplyr::rowwise() %>%
+  dplyr::mutate(num_resources_support = 
+                  length(stringr::str_split(support, 
+                                            pattern = ",", 
+                                            simplify = TRUE)))
+
+```
+
+To demonstrate how this works, we'll pick a gene that had 1:many mappings
+and follow it along.
+
+```{r}
+mouse_ortholog_df %>% 
+  dplyr::filter(human_entrez == 5631) %>%
+  dplyr::select(human_entrez, mouse_entrez, human_symbol, mouse_symbol,
+                num_resources_support)
+```
+
+We can see that the human gene _PRPS1_/`5631` maps to 4 mouse genes, one of 
+which has 10 resources supporting that mapping.
+
+```{r}
+# for each unique human entrez id, pick the mapping with the highest number of
+# resources supporting it
+most_support_df <- mouse_ortholog_df %>%
+  dplyr::group_by(human_entrez) %>%
+  dplyr::top_n(1, num_resources_support)
+```
+
+What happened to `5631`?
+
+```{r}
+most_support_df %>% 
+  dplyr::filter(human_entrez == 5631) %>%
+  dplyr::select(human_entrez, mouse_entrez, human_symbol, mouse_symbol,
+                num_resources_support)
+```
+
+We successfully selected the mapping with the highest number of resources
+supporting it.
+
+### Conversion of KEGG pathways
+
+For each KEGG pathway we have (currently populated with human Entrez IDs),
+we need a new gene set that is comprised of mouse gene symbols.
+
+```{r}
+kegg_mouse_list <- 
+  lapply(kegg_human_list, 
+         function(x) {
+           dplyr::filter(most_support_df, human_entrez %in% x) %>%
+             dplyr::pull(mouse_symbol)
+         })
+```
+
+Do the results this seem reasonable?
+Let's pick the polymerase pathway, where our success should be pretty obvious
+from the mouse gene symbols.
+
+```{r}
+kegg_mouse_list[[grep("POLYMERASE", names(kegg_mouse_list))]]
+```
+
+### Write mouse pathway list in `GMT` format
+
+We need to write this new mouse pathway list to file in [GMT format](http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29), as this will facilitate use with downstream pathway analysis.
+
+Briefly, the GMT format has one pathway per line and it follows this pattern:
+
+```
+<PATHWAY_NAME>\t<PATHWAY_DESCRIPTION>\t<gene_1_in_pathway>...<gene_n_in_pathway>
+```
+
+We've lost the description information because it's removed by 
+`qusage::read.gmt`.
+The description in `r kegg_file` follows this pattern:
```

Comment:
I don't exactly understand this mini section here (Lines 200 - 208). Perhaps if I ran it I would understand what you mean?

Reply:
I think the key point may be at line 232. As far as adding more context goes - how familiar would you say you are with the GMT file format?

Reply:
I see. 232 feels like the punchline here. I've only briefly worked with gmt files. Not much.

Reply:
Do you think this would make sense if you followed the link to GMT file formats?

Reply:
I looked at those. I'll just run the thing so I can see what the object looks like.

Reply:
You can also open the `.gmt` added here. I think that would clear things up. I would think most users would go that route? I'm trying to discern if there's additional context that would help, but I'm not sure there is.

Reply:
What does `r kegg_file`refer to? The website or an object in R?

Reply:
Discussed in person: In the HTML file, this would render as the full path assigned to `kegg_file`. This allows me to avoid hardcoding this path in the text.